keylime/da/examples/redis.py

```diff
@@ -0,0 +1,148 @@
+import importlib
+import os
+import re
+import time
+
+import redis
+
+from keylime import keylime_logging
+from keylime.da.record import BaseRecordManagement
+
+# setup logging
+logger = keylime_logging.init_logging("persistent_store")
+
+# ######################################################
+# Durable Attestation record manager with Redis backend
+# ######################################################
+
+
+class RecordManagement(BaseRecordManagement):
+    def __init__(self, service):
+
+        BaseRecordManagement.__init__(self, service)
+
+        self.redis_ip, self.redis_port, self.redis_db, self.redis_password, self.redis_prefix = re.split(
+            r":|\?|&",
+            self.ps_url.replace("redis://", "").replace("db=", "").replace("password=", "").replace("prefix=", ""),
+        )
+
+        self.redis_password = self.redis_password.replace("~", os.environ["HOME"])
+        self.redis_conn = None
+
+    def redis_connect(self):
+        """
+        TBD
+        """
+        if not self.redis_conn:
+            if os.path.exists(self.redis_password):
+
+                with open(self.redis_password, encoding="utf-8") as fp:
+                    self.redis_password = fp.read().strip()
```

Comment:
I see, so redis_password can be a file. I vote to use an env var, or other process. IIUC the passwords for other databases are stored in the config file, this different on treatment can be confusing

Reply:
Removed